src/cmds/openapi/uses.ts

```diff
@@ -0,0 +1,272 @@
+    ].forEach(({ component, header }: { component: 'openapi' | 'readme'; header: string }) => {
+      const tableData: string[][] = [
+        [chalk.bold.green('Feature'), chalk.bold.green('Used?'), chalk.bold.green('Description')],
+      ];
+
+      Object.entries(analysis[component]).forEach(([feature, info]) => {
+        const descriptions: string[] = [];
+        if (info.description) {
+          descriptions.push(info.description);
+        }
+
+        const url = this.getFeatureDocsURL(info);
+        if (url) {
+          descriptions.push(chalk.grey(url));
+        }
+
+        tableData.push([feature, info.present ? '✅' : '', descriptions.join('\n\n')]);
+      });
+
+      report.push('');
+      report.push(header);
+      report.push(
+        table(tableData, {
+          border: this.tableBorder,
+          columns: {
+            2: {
+              width: 80,
+              wrapWord: true,
+            },
+          },
+        })
+      );
+    });
+
+    return report.join('\n');
+  }
+
+  async run(opts: CommandOptions<Options>) {
+    await super.run(opts);
+
+    const { spec, workingDirectory, feature: features } = opts;
+
+    // If we have features we should validate that they're supported.
+    if (features) {
+      const invalidFeatures = features.filter(feature => !getSupportedFeatures().includes(feature));
+      if (invalidFeatures.length) {
+        return Promise.reject(
+          new Error(
+            `Unknown features: ${invalidFeatures.join(', ')}. See \`${config.get('cli')} help ${
+              this.command
+            }\` for help.`
+          )
+        );
+      }
+    }
+
+    if (workingDirectory) {
+      process.chdir(workingDirectory);
+    }
+
+    const { bundledSpec, definitionVersion } = await prepareOas(spec, 'openapi:uses', { convertToLatest: true });
+    this.definitionVersion = definitionVersion.version;
+    const parsedBundledSpec = JSON.parse(bundledSpec);
+
+    const spinner = ora({ ...oraOptions() });
+    if (features) {
+      spinner.start(
+        `Analyzing your API definition for usage of ${new Intl.ListFormat('en').format(
+          features.map(feature => (feature === 'readme' ? 'ReadMe extensions' : feature))
+        )}...`
+      );
+    } else {
+      spinner.start('Analyzing your API definition for OpenAPI and ReadMe feature usage...');
+    }
+
+    const analysis = await analyzeOas(parsedBundledSpec).catch(err => {
+      Command.debug(`analyzer err: ${err.message}`);
+      spinner.fail();
+      throw err;
+    });
+
+    spinner.succeed(`${spinner.text} done! ✅`);
+
+    if (features) {
+      const { report, hasUnusedFeature } = this.buildFeaturesReport(analysis, features);
+      if (hasUnusedFeature) {
+        // If we have any unused features we should reject the command with a soft error so we
+        // output the report as normal but return a `exit(1)` status code.
+        return Promise.reject(new SoftError(report));
```

Comment:
We talked about doing this as a `Promise.allSettled` but it was much easier to have `this.buildFeaturesReport` return a boolean for this instead of overhauling the already complex report composition to fit into the structure of a Promise.